ui/oopsyraidsy/data/03-hw/trial/sephirot-ex.ts

```diff
@@ -0,0 +1,152 @@
+  },
+  soloWarn: {
+    'SephirotEX Fiendish Rage': '156D', // Stack markers, phase 1
+  },
+  triggers: [
+    {
+      // Pillar of Mercy,  Malkuth, and Pillar of Severity
+      id: 'SephirotEX Knockbacks',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: ['1580', '1582', '1586'], source: 'Sephirot' }),
+      deathReason: (_data, matches) => {
+        return {
+          id: matches.targetId,
+          name: matches.target,
+          text: {
+            en: 'Pushed off!',
+            de: 'Runter geschubst!',
+            fr: 'Repoussé(e) !',
+            ja: '落ちた',
+            cn: '击退坠落',
+            ko: '넉백됨',
+          },
+        };
+      },
+    },
+    {
+      // 3ED is Force Against Might orange, 3EE is Force Against Magic, green.
+      id: 'SephirotEX Force Against Gain',
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: ['3ED', '3EE'] }),
+      run: (data, matches) => {
+        data.force ??= {};
+        data.force[matches.target] = matches.effectId;
+      },
+    },
+    {
+      id: 'SephirotEX Force Against Lose',
+      type: 'LosesEffect',
+      netRegex: NetRegexes.losesEffect({ effectId: ['3ED', '3EE'] }),
+      run: (data, matches) => {
+        data.force ??= {};
+        delete data.force[matches.target];
+      },
+    },
+    {
+      id: 'SephirotEX Spirit',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '157B', source: 'Sephirot' }),
+      condition: (data, matches) => data?.force?.[matches.target] === '3ED',
+      mistake: (_data, matches) => {
+        return {
+          type: 'warn',
+          blame: matches.target,
+          text: {
+            en: 'Took green; orange debuff',
+          },
+        };
+      },
+    },
+    {
+      id: 'SephirotEX Life Force',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '157A', source: 'Sephirot' }),
+      condition: (data, matches) => data?.force?.[matches.target] === '3EE',
+      mistake: (_data, matches) => {
+        return {
+          type: 'warn',
+          blame: matches.target,
+          text: {
+            en: 'Took orange; green debuff',
+          },
+        };
+      },
+    },
+    {
+      id: 'SephirotEX Fiendish Wail Green',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '1576', source: 'Sephirot' }),
+      condition: (data, matches) => data?.force?.[matches.target] === '3EE',
+      mistake: (_data, matches) => {
+        return {
+          type: 'warn',
+          blame: matches.target,
+          text: {
+            en: 'Tower with green',
+          },
+        };
+      },
+    },
+    {
+      // Only tanks should take towers without a Force debuff.
+      id: 'SephirotEX Fiendish Wail Non-Tank',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '1576', source: 'Sephirot' }),
+      condition: (data, matches) => !data.party.isTank(matches.target) && data?.force?.[matches.target] === undefined,
```

Comment:
I think for now just leave this as-is and assume that we'll (I'll?) fix this sometime soon.

Also, how should this handle BLU?

Reply:
Fair point, we can just not call Blue Mages if they take towers without a debuff. Fortunately that's the only role-based trigger we'll need for this encounter!